# **Caso de estudio**
Se quiere implementar un modelo de árbol de decisión para pronosticar si los pacientes
que hacen parte de esta base de datos van a ser diagnosticados con Alzheimer.
Para este proceso vamos a utilizar las variables del dataset:

**Age**: Edad del paciente en años.

**Gender**: Género del paciente (0 = Femenino, 1 = Masculino).

**Ethnicity**: Etnia del paciente (0 = Caucásico, 1 = Afroamericano, 2 = Asiático, 3 = Otro).

**EducationLevel**: Nivel educativo del paciente (0 = Ninguno, 1 = Secundaria, 2 = Universitario, 3 = Posgrado).

**BMI**: Índice de masa corporal del paciente.

**Smoking**: Indica si el paciente fuma (0 = No, 1 = Sí).

**AlcoholConsumption**: Consumo semanal de alcohol en unidades.

**PhysicalActivity**: Actividad física semanal en horas.

**DietQuality**: Puntuación de la calidad de la dieta del paciente (0-10).

**SleepQuality**: Puntuación de la calidad del sueño del paciente (4-10).

**FamilyHistoryAlzheimers**: Indica si hay antecedentes familiares de Alzheimer (0 = No, 1 = Sí).

**CardiovascularDisease**: Indica si el paciente tiene enfermedad cardiovascular (0 = No, 1 = Sí).

**Diabetes**: Indica si el paciente tiene diabetes (0 = No, 1 = Sí).

**Depression**: Indica si el paciente tiene depresión (0 = No, 1 = Sí).

**HeadInjury**: Indica si el paciente ha sufrido una lesión en la cabeza (0 = No, 1 = Sí).

**Hypertension**: Indica si el paciente tiene hipertensión (0 = No, 1 = Sí).

**SystolicBP**: Presión arterial sistólica del paciente (mmHg).

**DiastolicBP**: Presión arterial diastólica del paciente (mmHg).

**CholesterolTotal**: Nivel de colesterol total del paciente (mg/dL).

**CholesterolLDL**: Nivel de colesterol LDL del paciente (mg/dL).

**CholesterolHDL**: Nivel de colesterol HDL del paciente (mg/dL).

**CholesterolTriglycerides**: Nivel de triglicéridos del paciente (mg/dL).

**MMSE**: Puntuación del Mini-Mental State Examination. Evalúa la función cognitiva del paciente (0-30).

**FunctionalAssessment**: Evaluación de la capacidad funcional del paciente (0-10).

**MemoryComplaints**: Indica si el paciente reporta quejas de memoria (0 = No, 1 = Sí).

**BehavioralProblems**: Indica si el paciente presenta problemas de comportamiento (0 = No, 1 = Sí).

**ADL**: Puntuación de actividades de la vida diaria del paciente (0-10).

**Confusion**: Indica si el paciente presenta episodios de confusión (0 = No, 1 = Sí).

**Disorientation**: Indica si el paciente presenta desorientación (0 = No, 1 = Sí).

**PersonalityChanges**: Indica si el paciente presenta cambios de personalidad (0 = No, 1 = Sí).

**DifficultyCompletingTasks**: Indica si el paciente tiene dificultad para completar tareas (0 = No, 1 = Sí).

**Forgetfulness**: Indica si el paciente presenta olvidos frecuentes (0 = No, 1 = Sí).

0. Se procede con la carga de librerías de trabajo

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import confusion_matrix

1. Se procede con la carga de datos de trabajo.

In [2]:
nxl= '/content/Alzheimer_disease_dataset.xlsx'
XDB= pd.read_excel(nxl, sheet_name= 0)

#Para el modelo de árbol de decisión se seleccionaron las 5 variables con mayor
#correlación con el diagnóstico, identificadas previamente en el modelo Naive Bayes

XD1 = XDB[['FunctionalAssessment', 'ADL', 'MemoryComplaints', 'MMSE', 'BehavioralProblems']]
yd1 = XDB['Diagnosis']

display(XD1)
display(yd1)

,FunctionalAssessment,ADL,MemoryComplaints,MMSE,BehavioralProblems
0,6.518877,1.725883,0,21.463532,0
1,7.118696,2.592424,0,20.613267,0
2,5.895077,7.119548,0,7.356249,0
3,8.965106,6.481226,0,13.991127,1
4,6.045039,0.014691,0,13.517609,0
...,...,...,...,...,...
2144,0.238667,4.492838,0,1.201190,0
2145,8.687480,9.204952,0,6.458060,1
2146,1.972137,5.036334,0,17.011003,0
2147,5.173891,3.785399,0,4.030491,0


,Diagnosis
0,0
1,0
2,0
3,0
4,0
...,...
2144,1
2145,1
2146,1
2147,1


2. Se procede con la implementación del modelo de árbol

In [3]:
mar=DecisionTreeClassifier(criterion='gini', max_depth=4) #Entrenamos el árbol con base histórica de datos
mar.fit(XD1,yd1)

ydp1=mar.predict(XD1) #Predicción del modelo
#Se construye la matriz de confusión
cm=confusion_matrix(yd1,ydp1)
display(cm)
VN=cm[0,0]; FP=cm[0,1]; FN=cm[1,0]; VP=cm[1,1]

#Métricas de desempeño
Ex=(VP+VN)/len(XD1) #1. Exactitud: Comportamiento general
print('La Exactitud es: ',Ex)
Sen=VP/(VP+FN) #2. Sensibilidad: Como se comporta pronosticando PreAprobados
print('La Sensibilidad es: ',Sen)
Spe=VN/(VN+FP) #3. Especificidad: Como se comporta frente a los negativos
print('La Especificidad es: ',Spe)
Pre=VP/(VP+FP) #4. Precisión: Como se comporta pronosticando PreAprobados
print('La Precisión es: ',Pre)
PreNeg=VN/(VN+FN) #5. Precisión Negativa: Como se comporta pronosticando Negativos
print('La Precisión Negativa es: ',PreNeg)

array([[1335,   54],
       [ 101,  659]])

La Exactitud es:  0.927873429502094
La Sensibilidad es:  0.8671052631578947
La Especificidad es:  0.9611231101511879
La Precisión es:  0.9242636746143057
La Precisión Negativa es:  0.9296657381615598


**Análisis de resultados**

De la base de datos histórica se observa un total de 2149 registros, de los cuales
1389 corresponden a pacientes sin diagnóstico de Alzheimer y 760 corresponden a
pacientes con diagnóstico de Alzheimer.

De los 1389 casos sin Alzheimer, el modelo clasificó correctamente 1335 pacientes,
mientras que de los 760 casos con Alzheimer, el modelo identificó correctamente 659
registros. El modelo logra identificar en un 96.1% a los pacientes sin Alzheimer
(1335/1389), mientras que logra identificar en un 86.7% a los pacientes con Alzheimer
(659/760).

Con respecto a las métricas, el modelo alcanzó una exactitud del 92.8%, lo que indica
un excelente comportamiento general del modelo frente a la clasificación en ambas
categorías. Se destacan especialmente la especificidad (96.1%) y la precisión negativa
(93.0%), las cuales se ubican ampliamente por encima del umbral del 75% definido como
referencia para este tipo de modelos de clasificación. La sensibilidad (86.7%) también
presenta un desempeño sólido, mostrando una adecuada capacidad para detectar pacientes
con diagnóstico de Alzheimer. La precisión (92.4%) evidencia que cuando el modelo
clasifica a un paciente como con Alzheimer, en la gran mayoría de los casos el
diagnóstico es correcto.

Es importante destacar que este modelo, construido únicamente con las 5 variables más
determinantes (FunctionalAssessment, ADL, MemoryComplaints, MMSE y BehavioralProblems),
supera en todas las métricas al modelo Naive Bayes que utilizó las 32 variables
disponibles, lo que confirma que estas variables capturan de manera efectiva los
patrones de clasificación del diagnóstico de Alzheimer.


3. Despliegue del árbol de decisión

In [4]:
from sklearn.tree import export_graphviz #Exporta los datos a un gráfico
from pydotplus import graph_from_dot_data #Es un graficador

vs=['FunctionalAssessment', 'ADL', 'MemoryComplaints', 'MMSE', 'BehavioralProblems'] #Títulos del árbol
dot_data=export_graphviz(mar, feature_names=vs) #Exportar de números a gráfico en pdf
graph=graph_from_dot_data(dot_data) #Graficamo el árbol
graph.write_png('arbol_alzheimer.png') #Guardamos el árbol en pdf

True

**Análisis de resultados**
De acuerdo con el árbol de decisión, la variable raíz es FunctionalAssessment con un
umbral de 5.001, dividiendo los 2149 pacientes en dos grandes grupos: 1054 pacientes
con baja capacidad funcional (rama izquierda) y 1095 con alta capacidad funcional
(rama derecha), lo que confirma que esta variable es el factor más determinante para
la clasificación del diagnóstico de Alzheimer.

En la rama derecha del árbol se identifica un nodo puro con value = [24, 0], donde el 100% de los individuos son
clasificados como sin Alzheimer, evidenciando un perfil de bajo riesgo claramente
diferenciado. Así mismo, se destaca el nodo casi puro con value = [710, 27] con un gini de 0.071, donde la gran
mayoría de los individuos no presentan diagnóstico de Alzheimer.

En la rama izquierda se destaca el nodo casi puro con value = [158, 3] y gini = 0.037, donde casi la totalidad
de los individuos no presentan Alzheimer. Por otro lado, el nodo con value = [2, 5]
y gini = 0.408, concentra la mayor proporción de casos positivos, evidenciando
que la combinación de baja capacidad funcional, dificultad en actividades diarias
y quejas de memoria constituye el perfil de mayor riesgo para el diagnóstico
de Alzheimer.

4. Pronósticar con 'Datos Pronóstico'

In [5]:
#Paciente 1 → [4.0, 4.0, 1, 18, 1]
#Paciente 2 → [8.0, 9.0, 0, 27, 0]
#Paciente 3 → [5.5, 5.0, 1, 22, 1]

XD2 = pd.DataFrame({
    'FunctionalAssessment': [4.0, 8.0, 5.5],
    'ADL':                  [4.0, 9.0, 5.0],
    'MemoryComplaints':     [1,   0,   1  ],
    'MMSE':                 [18,  27,  22 ],
    'BehavioralProblems':   [1,   0,   1  ]
})

ydp2 = mar.predict(XD2)
display(ydp2)

for i, pred in enumerate(ydp2):
    if pred == 1:
        print(f"Paciente {i+1}: El paciente tiene diagnóstico positivo de Alzheimer")
    else:
        print(f"Paciente {i+1}: El paciente no presenta diagnóstico de Alzheimer")

array([1, 0, 1])

Paciente 1: El paciente tiene diagnóstico positivo de Alzheimer
Paciente 2: El paciente no presenta diagnóstico de Alzheimer
Paciente 3: El paciente tiene diagnóstico positivo de Alzheimer


**Análisis Pronóstico**

De los 3 pacientes evaluados, el modelo pronostica que 1 paciente (1/3 = 33.3%)
no presenta diagnóstico de Alzheimer, mientras que 2 pacientes (2/3 = 66.7%)
tienen diagnóstico positivo.
